dpsim-models/include/dpsim-models/Solver/EigenvalueCompInterface.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+#include <dpsim-models/Definitions.h>
+
+namespace CPS {
+/// Interface to be implemented by all components taking part in eigenvalue extraction.
+class EigenvalueCompInterface {
+public:
+  using Ptr = std::shared_ptr<EigenvalueCompInterface>;
+  using List = std::vector<Ptr>;
```

Comment:
This is done in the other component interfaces as well, but generally I think it is better to be explicit about data types. It might not be clear that EigenvalueCompInterface::List is a vector of shared pointers of exactly the class the type was defined in.

It might be hard to understand since EigenvalueCompInterface::List could be any "list"-type containing anything within the interface class. However, once you do know what it means and use it often it may improve readability. I still vote against it though personally.

Reply:
thanks for your feedback! I agree with your arguments. In this particular case even the actual type would not read too bad `std::vector<std::shared_ptr<EigenvalueCompInterface>>`. However, if there were no aliases at all, codebase in general would become less readable. So, for now let's just keep consistent with the other component interfaces.